reth-extension/src/pool/pool.rs

```diff
@@ -186,21 +186,43 @@ where
         let (_, tx) = self.validate(origin, transaction).await;
         self.pool.add_transaction_and_subscribe(origin, tx)
     }
+
     /*
      * Scalaris
-     * Todo: Inject abci client to send transaction out
+     * Inject abci client to send transaction out
      */
     async fn add_transaction(
         &self,
         origin: TransactionOrigin,
         transaction: Self::Transaction,
     ) -> PoolResult<TxHash> {
-        let (_, tx) = self.validate(origin, transaction).await;
-        let mut results = self.pool.add_transactions(origin, std::iter::once(tx));
-        results
-            .pop()
-            .expect("result length is the same as the input")
+        use crate::abci::abci_ext::AbciMessage;
+        use reth_primitives::{IntoRecoveredTransaction, TransactionSigned};
+
+        let dummy_pool_result = PoolResult::Ok(TxHash::default());
+
+        if origin != TransactionOrigin::Local {
```

Comment:
the whole logic here can be rewritten better like followings:


        if origin != TransactionOrigin::Local {
            // Only process local transactions
            return PoolResult::Ok(TxHash::default());
        }


        if let Some(tx_abci) = &self.tx_abci {
            let recovered_transaction = transaction.to_recovered_transaction();
            let signed_transaction = TransactionSigned::from_transaction_and_signature(
                recovered_transaction.transaction.clone(),
                recovered_transaction.signature,
            );
            let tx_bytes = signed_transaction.envelope_encoded()[..].to_vec();

            // Send transaction to abci adapter
            let tx = AbciMessage::RawTransaction(tx_bytes);
            tx_abci.send(tx).await.unwrap();
        }

        // Return dummy pool result to end the transaction flow instead of continuing Reth logic
        PoolResult::Ok(TxHash::default());